File: make-admin.py
#!/usr/bin/env python3
import os
import sys
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def make_admin(email):
    try:
        # Connect to MongoDB
        mongo_uri = os.getenv('MONGODB_URI')
        if not mongo_uri:
            print("❌ MONGODB_URI not found in .env file")
            return False

        client = MongoClient(mongo_uri, serverSelectionTimeoutMS=5000)
        db = client.get_default_database()

        print('Connected to MongoDB\n')

        # Find user
        user = db.users.find_one({'email': email.lower()})

        if not user:
            print(f"❌ User not found with email: {email}")
            print("\nMake sure the user has an instructor account first.")
            return False

        print(f"Found user: {user.get('firstName', '')} {user.get('lastName', '')}")
        print(f"Current role: {user.get('role', 'instructor')}")

        if user.get('role') == 'admin':
            print("\n✅ User is already an admin!")
            return True

        # Update to admin
        result = db.users.update_one(
            {'email': email.lower()},
            {'$set': {'role': 'admin'}}
        )

        if result.modified_count > 0:
            print(f"\n✅ Successfully updated {email} to admin role!")
            print("\nYou can now access the admin panel at:")
            print("https://instructor.intellaclick.com/admin.html")
            return True
        else:
            print("❌ Failed to update user")
            return False

    except Exception as e:
        print(f"Error: {str(e)}")
        return False
    finally:
        if 'client' in locals():
            client.close()

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print('Usage: python3 make-admin.py <email>')
        print('Example: python3 make-admin.py mostafa.afifi77@gmail.com')
        sys.exit(1)

    email = sys.argv[1]
    success = make_admin(email)
    sys.exit(0 if success else 1)

File: make-admin.js
// Script to make a user an admin
// Usage: node make-admin.js <email>

const mongoose = require('mongoose');
require('dotenv').config();

const User = require('./models/User');

async function makeAdmin(email) {
  try {
    await mongoose.connect(process.env.MONGODB_URI);
    console.log('Connected to MongoDB\n');

    const user = await User.findOne({ email: email.toLowerCase() });

    if (!user) {
      console.log(`❌ User not found with email: ${email}`);
      console.log('\nMake sure the user has an instructor account first.');
      process.exit(1);
    }

    console.log(`Found user: ${user.firstName} ${user.lastName}`);
    console.log(`Current role: ${user.role}`);

    if (user.role === 'admin') {
      console.log('\n✅ User is already an admin!');
      process.exit(0);
    }

    // Update to admin
    user.role = 'admin';
    await user.save();

    console.log(`\n✅ Successfully updated ${email} to admin role!`);
    console.log('\nYou can now access the admin panel at:');
    console.log('https://instructor.intellaclick.com/admin.html');

    mongoose.disconnect();
  } catch (error) {
    console.error('Error:', error.message);
    mongoose.disconnect();
    process.exit(1);
  }
}

// Get email from command line
const email = process.argv[2];

if (!email) {
  console.log('Usage: node make-admin.js <email>');
  console.log('Example: node make-admin.js mostafa.afifi77@gmail.com');
  process.exit(1);
}

makeAdmin(email);
